File: tests/human/x/ts/cross_join_triple.ts
const nums = [1, 2];
const letters = ["A", "B"];
const bools = [true, false];
const combos = [] as Array<{ n: number; l: string; b: boolean }>;
for (const n of nums) {
  for (const l of letters) {
    for (const b of bools) {
      combos.push({ n, l, b });
    }
  }
}
console.log("--- Cross Join of three lists ---");
for (const c of combos) {
  console.log(c.n, c.l, c.b);
}

File: tests/human/x/kt/cross_join_triple.kt
fun main() {
    val nums = listOf(1, 2)
    val letters = listOf("A", "B")
    val bools = listOf(true, false)
    val combos = mutableListOf<Triple<Int, String, Boolean>>()
    for (n in nums) {
        for (l in letters) {
            for (b in bools) {
                combos.add(Triple(n, l, b))
            }
        }
    }
    println("--- Cross Join of three lists ---")
    for (c in combos) {
        println("${c.first} ${c.second} ${c.third}")
    }
}
